### DISCUSSION TITLE
How to couple vectorpostprocessor in header & source files ?

### DISCUSSION THREAD

**Discussion by Minjiang-Zhu** [idaholab/moose#25841] [Q&A General] [answered]
As mentioned in title, I wonder if there is any instruction on how to couple vectorpostprocessor.

**Comment by Minjiang-Zhu** [comment]
Seems there are several files in framework/(include or source)/vectorpostprocessors. I'll imitate the coding style and see if it shall work.

**Comment by GiudGiud** [comment, marked as answer]
Most objects inherit VectorPostprocessorInterface from which you can call getVectorPostprocessorXYZ to retrieve whatever you want. 

Most likely you want the value

**Reply by Minjiang-Zhu** [reply]
Thank you! I imitated the codes in "LeastSquareFit", referenced [this site](https://mooseframework.inl.gov/syntax/VectorPostprocessors/)  and obtained what I want.
